morebits.js

```diff
@@ -1221,16 +1221,23 @@ Morebits.string = {
 	 * Formats freeform "reason" (from a textarea) for deletion/other
 	 * templates that are going to be substituted, (e.g. PROD, XFD, RPP).
 	 * Handles `|` outside a nowiki tag.
+	 * Optionally, also adds a signature if not present already.
 	 *
 	 * @param {string} str
+	 * @param {boolean} addSig
 	 * @returns {string}
 	 */
-	formatReasonText: function(str) {
-		var result = str.toString().trim();
-		var unbinder = new Morebits.unbinder(result);
+	formatReasonText: function(str, addSig) {
+		var reason = (str || '').toString().trim();
+		var unbinder = new Morebits.unbinder(reason);
 		unbinder.unbind('<no' + 'wiki>', '</no' + 'wiki>');
 		unbinder.content = unbinder.content.replace(/\|/g, '{{subst:!}}');
-		return unbinder.rebind();
+		reason = unbinder.rebind();
+		var sigIndex = reason.lastIndexOf('~~~~');
+		if (addSig && (sigIndex === -1 || sigIndex !== reason.length - 4)) {
```

Comment:
ES5 equivalent of `endsWith`, ugh

Reply:
Two lines longer but a little prettier if you move up `addSig` above `sigIndex`?

Could also do `sig = '~~~~'`, `lastIndexOf(sig)`, and `reason.length - sig.length`, but I'm sure you considered it.

Reply:
Ah yeah that's better. Didn't consider it.